### DISCUSSION TITLE
Cannot be executed on Alpine

### DISCUSSION THREAD

**Discussion by gridhead** [coder/code-server#2638] [Q&A]
Hi folks, `code-server` fails to run on an Alpine container when a `./code-server` command is executed. It fails to find the `node` executable which is present at the correct relative directory location. Any help would be appreciated.

(Does this has to do with the `musl` vs `glibc` thingy? I am curious.)

**Comment by code-asher** [comment]
Yup, that's why. I'm not sure if it's possible to use glibc on Alpine
but one way to use code-server would be to install it with yarn/npm
which will build it for your system:

https://github.com/cdr/code-server/blob/v3.8.0/doc/npm.md#alpine

**Comment by gridhead** [comment]
I see.

In contrast to the Docker images for `code-server` built by `linuxserver.io` which uses Debian as the base, would it be feasible to follow the instructions provided at https://github.com/cdr/code-server/blob/v3.8.0/doc/npm.md#alpine to build a Docker image based on Alpine then?

**Reply by code-asher** [reply]
Yep, that should work. Alternatively you can follow the build instructions and build from source, but I think building from npm is going to be a bit easier.

**Reply by martinussuherman** [reply]
> 
> 
> I see.
> 
> In contrast to the Docker images for `code-server` built by `linuxserver.io` which uses Debian as the base, would it be feasible to follow the instructions provided at https://github.com/cdr/code-server/blob/v3.8.0/doc/npm.md#alpine to build a Docker image based on Alpine then?

I got alpine build (which I've deployed to a few server), you can find it on my repo: https://github.com/martinussuherman/alpine-code-server. Hope that helps.

Regards,

Martinus

**Reply by gridhead** [reply]
Thanks, @martinussuherman. This image is really lightweight in terms of size, albeit the absence of multiarchitecture versions is really hurting here. Would you be willing to make `arm64/armv7` images down the line? I can be of assistance if you are interested.

@code-asher We can incorporate the build configuration that @martinussuherman has come up with in the documentation as many folks are asking the same question but in the issues section.

**Reply by martinussuherman** [reply]
> 
> 
> Thanks, @martinussuherman. This image is really lightweight in terms of size, albeit the absence of multiarchitecture versions is really hurting here. Would you be willing to make `arm64/armv7` images down the line? I can be of assistance if you are interested.
> 
> @code-asher We can incorporate the build configuration that @martinussuherman has come up with in the documentation as many folks are asking the same question but in the issues section.

I could help create those images if you want to. Please help me of anything I should change on my Dockerfile.
Fyi, my image was based on this:  https://github.com/jeanblanchard/docker-alpine-glibc so this one should be built for arm64/armv7 too. 

Regards,

Martinus

**Reply by bpmct** [reply]
Awesome image @martinussuherman. I've added this to [awesome-code-server](https://github.com/bpmct/awesome-code-server).

**Reply by martinussuherman** [reply]
Thanks for your favor @bpmct.

**Comment by gridhead** [comment]
> > Thanks, @martinussuherman. This image is really lightweight in terms of size, albeit the absence of multiarchitecture versions is really hurting here. Would you be willing to make `arm64/armv7` images down the line? I can be of assistance if you are interested.
> > @code-asher We can incorporate the build configuration that @martinussuherman has come up with in the documentation as many folks are asking the same question but in the issues section.
> 
> I could help create those images if you want to. Please help me of anything I should change on my Dockerfile.
> Fyi, my image was based on this: https://github.com/jeanblanchard/docker-alpine-glibc so this one should be built for arm64/armv7 too.
> 
> Regards,
> 
> Martinus

Hey @martinussuherman,

I will take your `Dockerfile` for a spin on my Raspberry Pi 3 and 4 to see if they can be supported on `armv7`/`aarch64` out-of-the-box as soon as my area recovers from a network outage. I will keep you posted about it.

We would want to open up a new discussion thread for this though as the discussion has met its solution here https://github.com/cdr/code-server/discussions/2638#discussioncomment-422325. What do you think?

@code-asher @bpmct Can we use the discussions forum here for the conversation as it is relevant to Coder?

**Reply by bpmct** [reply]
> Can we use the discussions forum here for the conversation as it is relevant to Coder?

Totally! Feel free to share any projects you make with the "Show and Tell" tag too in discussions :)

**Reply by gridhead** [reply]
Thanks, @bpmct.

I do have some projects that I want to show to folks and know what they think about it, albeit they might not be completely related to Coder. Can I still post them? :slightly_smiling_face:

**Reply by ChillySpark** [reply]
@martinussuherman 

I used one of your prior images I found on Docker Hub to run a container on AWS EC2 that I access from my iPad Pro. It works great. I also used your latest image to run a container on Windows 10 under WSL... also great! I hope to try it out on AWS ECS (Fargate Spot) soon. Glibc also enables an AWSCLI v2 installation. Thanks so much to you and everyone contributing to Code-Server!

**Reply by martinussuherman** [reply]
> Hey @martinussuherman,
> 
> I will take your `Dockerfile` for a spin on my Raspberry Pi 3 and 4 to see if they can be supported on `armv7`/`aarch64` out-of-the-box as soon as my area recovers from a network outage. I will keep you posted about it.

@t0xic0der great, please let me know the result.

> We would want to open up a new discussion thread for this though as the discussion has met its solution here [#2638 (reply in thread)](https://github.com/cdr/code-server/discussions/2638#discussioncomment-422325). What do you think?
 
No problem for me.

@GregPriceDev thank you for your compliment.

Regards,

Martinus

**Reply by RyanTheDevGuy** [reply]
@GregPriceDev did you ever get Coder running on Fargate?
